File: packages/duit_js/src/widget_models/text.ts
import DuitElementType from "../lib/element_type";
import { DuitLeafElement } from "./child";

import type { TextAttributes } from "../attributes";
import { BaseAction } from "../lib/action";

export class TextUiElement extends DuitLeafElement {
  type = DuitElementType.text as const;
  attributes: TextAttributes;

  constructor(attrs: TextAttributes, id?: string, action?: BaseAction, controlled?: boolean) {
    super(id, action, controlled);
    this.attributes = attrs;
  }
}

File: packages/duit_go/pkg/duit_widget/text.go
package duit_widget

import (
	attrs "github.com/lesleysin/duit/packages/duit_go/pkg/duit_attributes"
	core "github.com/lesleysin/duit/packages/duit_go/pkg/duit_core"
)

func TextUiElement(attributes *attrs.TextAttributes, id string, controlled bool, action *core.Action) *core.DuitElementModel {
	return new(core.DuitElementModel).CreateElement(core.Text, id, "", attributes, action, controlled)
}
